lib/idtree.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * id allocating tree
+ *
+ * Copyright 2018 Phoenix Systems
+ * Author: Jan Sikorski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "lib.h"
+
+
+static int lib_idtreeCmp(rbnode_t *n1, rbnode_t *n2)
+{
+	idnode_t *i1 = lib_treeof(idnode_t, linkage, n1);
+	idnode_t *i2 = lib_treeof(idnode_t, linkage, n2);
+
+	if (i1->id > i2->id) {
+		return 1;
+	}
+	else if (i2->id > i1->id) {
+		return -1;
+	}
+	else {
+		return 0;
+	}
+}
+
+
+static int lib_idtreeGapcmp(rbnode_t *n1, rbnode_t *n2)
+{
+	idnode_t *r1 = lib_treeof(idnode_t, linkage, n1);
+	idnode_t *r2 = lib_treeof(idnode_t, linkage, n2);
+	rbnode_t *child = NULL;
+	int ret = 1;
```

Comment:
IMHO `ret` should be set to zero as that's its de facto default value, or not set at all as whenevere child != NULL then ret is set

Reply:
As it turns out we don't need a child at all

Reply:
Retreat, it's need afterall